### PR TITLE
Fix macOS CI tests

### PR DESCRIPTION
**Pull request by jezdez**
- Switch to macos-13 as a work-around for the new macos-latest not providing a default miniconda installation anymore
- add macOS on ARM64 testing
- add conda 24.3.0 to testing

### CODE REVIEW

**Review by dbast — APPROVED** [review]

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -46,12 +46,13 @@ jobs:
...
       run:
-        shell: bash
+        # https://github.com/conda-incubator/setup-miniconda#use-a-default-shell
+        shell: bash -el {0}  # bash exit immediately on error + login shell
     strategy:
       fail-fast: false
       matrix:
-        os: [ubuntu-latest, macos-latest, windows-latest]
-        cversion: ['4.11.0', '4.14.0', '22.11.1', '23.3.1', '23.7.3', '23.10.0', '24.1.2']
+        os: [ubuntu-latest, macos-13, windows-latest]
```
Comment (jezdez) [inline .github/workflows/main.yml:54]: work-around for miniconda missing in latest macos runner images

File: .github/workflows/main.yml [hunk 2]
```diff
@@ -46,12 +46,13 @@ jobs:
...
+        cversion: ['4.11.0', '4.14.0', '22.11.1', '23.3.1', '23.7.3', '23.10.0', '24.1.2', '24.3.0']
```
Comment (jezdez) [inline .github/workflows/main.yml:55]: adding the latest version of conda

File: .github/workflows/main.yml [hunk 3]
```diff
@@ -63,12 +64,17 @@ jobs:
       with:
         name: conda-bld
         path: conda-bld
+    - name: Setup Miniconda
+      uses: conda-incubator/setup-miniconda@a4260408e20b96e80095f42ff7f1a15b27dd94ca
+      with:
+        miniconda-version: 'latest'
+        auto-activate-base: true
+        activate-environment: ""
+        conda-solver: classic
```
Comment (jezdez) [inline .github/workflows/main.yml:73]: classic since that would make it work with the older versions of conda that are being tested

**Review by mcg1969 — APPROVED** [review]

File: .github/workflows/main.yml [hunk 4]
```diff
@@ -63,12 +65,17 @@ jobs:
       with:
         name: conda-bld
         path: conda-bld
+    - name: Setup Miniconda
+      uses: conda-incubator/setup-miniconda@a4260408e20b96e80095f42ff7f1a15b27dd94ca
```
Comment (dbast) [inline .github/workflows/main.yml:69]: Please keep the real version as comment... both dependabot / renovate also update the version in comment
```suggestion
      uses: conda-incubator/setup-miniconda@a4260408e20b96e80095f42ff7f1a15b27dd94ca # v3.0.4
```

Comment (jezdez) [reply .github/workflows/main.yml]: Thanks!

**Review by mcg1969 — APPROVED** [review]

**Pull request merged**